core/src/main/scala/kafka/network/RequestConvertToJson.scala

```diff
@@ -0,0 +1,343 @@
+      case req: FindCoordinatorRequest => FindCoordinatorRequestDataJsonConverter.write(req.data(), request.version())
+      case req: HeartbeatRequest => HeartbeatRequestDataJsonConverter.write(req.data(), request.version())
+      case req: IncrementalAlterConfigsRequest => IncrementalAlterConfigsRequestDataJsonConverter.write(req.data(), request.version())
+      case req: InitProducerIdRequest => InitProducerIdRequestDataJsonConverter.write(req.data(), request.version())
+      case req: JoinGroupRequest => JoinGroupRequestDataJsonConverter.write(req.data(), request.version())
+      case req: LeaderAndIsrRequest => LeaderAndIsrRequestDataJsonConverter.write(req.data(), request.version())
+      case req: LeaveGroupRequest => LeaveGroupRequestDataJsonConverter.write(req.data(), request.version())
+      case req: ListGroupsRequest => ListGroupsRequestDataJsonConverter.write(req.data(), request.version())
+      case req: ListOffsetRequest => ListOffsetRequestDataJsonConverter.write(req.data(), request.version())
+      case req: ListPartitionReassignmentsRequest => ListPartitionReassignmentsRequestDataJsonConverter.write(req.data(), request.version())
+      case req: MetadataRequest => MetadataRequestDataJsonConverter.write(req.data(), request.version())
+      case req: OffsetCommitRequest => OffsetCommitRequestDataJsonConverter.write(req.data(), request.version())
+      case req: OffsetDeleteRequest => OffsetDeleteRequestDataJsonConverter.write(req.data(), request.version())
+      case req: OffsetFetchRequest => OffsetFetchRequestDataJsonConverter.write(req.data(), request.version())
+      case req: OffsetsForLeaderEpochRequest => offsetsForLeaderEpochRequestNode(req, request.version())
+      case req: ProduceRequest => produceRequestNode(req, request.version(), verbose)
+      case req: RenewDelegationTokenRequest => RenewDelegationTokenRequestDataJsonConverter.write(req.data(), request.version())
+      case req: SaslAuthenticateRequest => SaslAuthenticateRequestDataJsonConverter.write(req.data(), request.version())
+      case req: SaslHandshakeRequest => SaslHandshakeRequestDataJsonConverter.write(req.data(), request.version())
+      case req: StopReplicaRequest => StopReplicaRequestDataJsonConverter.write(req.data(), request.version())
+      case req: SyncGroupRequest => SyncGroupRequestDataJsonConverter.write(req.data(), request.version())
+      case req: TxnOffsetCommitRequest => TxnOffsetCommitRequestDataJsonConverter.write(req.data(), request.version())
+      case req: UpdateFeaturesRequest => UpdateFeaturesRequestDataJsonConverter.write(req.data(), request.version())
+      case req: UpdateMetadataRequest => UpdateMetadataRequestDataJsonConverter.write(req.data(), request.version())
+      case req: VoteRequest => VoteRequestDataJsonConverter.write(req.data, request.version())
+      case req: WriteTxnMarkersRequest => WriteTxnMarkersRequestDataJsonConverter.write(req.data(), request.version())
+    }
```

Comment:
There was an issue before where a request type wasn't handled and it threw a `MatchError`. I added the system test to make sure all the request types were handled, but in case this happens again, I'll add the exception for unknown cases.